ext-src/liveShare.ts

```diff
@@ -0,0 +1,136 @@
+import * as vsls from 'vsls/vscode';
+import { BrowserViewWindowManager, BrowserViewWindow, PANEL_TITLE } from './extension';
+
+const SERVICE_NAME = 'browser-preview';
+const REQUEST_GET_WINDOWS = 'getWindows';
+const NOTIFICATION_WINDOW_CREATED = 'windowCreated';
+const NOTIFICATION_WINDOW_INTERACTION = 'windowInteraction';
+
+const DISPATCHED_EVENTS = ['Input.dispatchMouseEvent', 'Input.dispatchKeyEvent'];
+const SHARED_PANEL_TITLE = `${PANEL_TITLE} (Shared)`;
+
+export async function setupLiveShare(windowManager: BrowserViewWindowManager) {
+  const liveShare = await vsls.getApi();
+  if (!liveShare) {
+    // The user doesn't have the Live Share
+    // extension installed, so there's nothing
+    // further to initialize.
+    return;
+  }
+
+  await sessionStarted(liveShare);
+  setupServices(liveShare, windowManager);
+}
+
+async function setupServices(liveShare: vsls.LiveShare, windowManager: BrowserViewWindowManager) {
+  let service: vsls.SharedService | vsls.SharedServiceProxy;
+
+  if (liveShare.session.role === vsls.Role.Host) {
+    service = (await liveShare.shareService(SERVICE_NAME))!;
+
+    service.onRequest(REQUEST_GET_WINDOWS, () => {
+      const windows = Array.from(windowManager.openWindows);
+      return windows.map((window) => {
+        const viewport = window.browserPage!.page.viewport();
+
+        return {
+          startUrl: window.config.startUrl, // TODO: This needs to be the current URL
+          viewport
+        };
+      });
+    });
+  } else if (liveShare.session.role === vsls.Role.Guest) {
+    service = await liveShare.getSharedService(SERVICE_NAME);
+
+    if (!service) {
+      // The host doesn't have this extension installed
+      return;
+    }
+
+    const windows = await service.request(REQUEST_GET_WINDOWS, []);
+    if (windows && windows.length > 0) {
+      windows.forEach(({ startUrl, viewport }: any) => {
+        windowManager.create(startUrl, SHARED_PANEL_TITLE);
```

Comment:
@auchenberg Once we have the dimensions of the host's preview window, what is the correct way is locking the guest's preview window to that size?

Reply:
We don't have such a way yet. And I don't believe we have a way to resize an WebviewPanel from an API. 

The solution would be to land something like https://github.com/auchenberg/vscode-browser-preview/issues/37

Reply:
Ah right, #37 sounds like exactly what this PR will need. Consider the Live Share integration to be another +1 to that feature request 😁